From d5d8a177244b0dabb4eb7ccbd1e735ffea93e875 Mon Sep 17 00:00:00 2001
From: Colin <colin@uninsane.org>
Date: Tue, 25 Aug 2026 01:16:23 +0000
Subject: [PATCH] alsa-utils: fix cross-compiled shebangs

before:
  ```
  $ head -n1 $(nix-build -A pkgsCross.aarch64-multiplatform.alsa-utils)/bin/.alsa-info-wrapped
  #!/bin/bash
  ```

and the resulting `alsa-info` would fail when invoked: "/bin/bash: bad interpreter: No such file or directory"

after:
  ```
  $ head -n1 $(nix-build -A pkgsCross.aarch64-multiplatform.alsa-utils)/bin/.alsa-info-wrapped
  #!/nix/store/p6mqrxz3wl8fvqcqqf498nyp88lv8687-bash-aarch64-unknown-linux-gnu-5.3p15/bin/bash
  ```
---
 pkgs/by-name/al/alsa-utils/package.nix | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pkgs/by-name/al/alsa-utils/package.nix b/pkgs/by-name/al/alsa-utils/package.nix
index 91653a4848b628a4aa51f629552a38826da9cd6d..4ad614132da542ae55b7ca761cb0b35a5c31cf38 100644
--- a/pkgs/by-name/al/alsa-utils/package.nix
+++ b/pkgs/by-name/al/alsa-utils/package.nix
@@ -5,6 +5,7 @@
   fetchurl,
   alsa-lib,
   alsa-plugins,
+  bashNonInteractive,
   gettext,
   makeWrapper,
   pkg-config,
@@ -45,6 +46,7 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
   ];
   buildInputs = [
+    bashNonInteractive
     alsa-lib
     ncurses
     libsamplerate
